### ISSUE TITLE
MPI_WIN_BASE and other MPI-2.2 macros not defined in AMPI

### ISSUE THREAD

**Issue by ericjbohm** [UIUC-PPL/charm#942]
*Original issue: https://charm.cs.illinois.edu/redmine/issues/942*

---
Per Jeff Hammond's (jeff.science`gmail.com) email

```
I am using Charm++ 6.7.0 for testing of the PRK.  I recently added code that fetches the MPI_WIN_BASE attribute on RMA windows.  This attribute was introduced in MPI 2.0, so it is not new.

Adaptive MPI does not define MPI_WIN_BASE, as demonstrate in https://travis-ci.org/jeffhammond/PRK/jobs/101958477.  I've already pushed a workaround (https://github.com/jeffhammond/PRK/commit/a511033a10995594bb8c90a6b0cdce6077bd3347), but it leads to a memory leak (this doesn't really matter, since the program terminates almost immediately thereafter, but it is shameful programming to do such things intentionally).

In short, all your MPI_WIN_BASE do not belong to us :-) <groan>

And if Adaptive MPI is setting MPI_VERSION=3 (I have not looked), then it needs the window attributes added in MPI 3.0.

```

**Comment by stwhite91** [comment]
*Original date: 2016-01-13 16:22:14*

---
~~https://charm.cs.illinois.edu/gerrit/#/c/998/~~ https://github.com/UIUC-PPL/charm/commit/e2e43465e3fd61597e674d31c2e8083fb82b2960
